main.py

```
@@ -148,37 +169,80 @@ def add_row(self, row_id: str, values: list[Any], dynamic_indices: Dict[str, int
+        if self._thread:
+            self._thread.join(timeout=2.0)
+
+    def _loop(self) -> None:
+        while not self._stop_event.is_set():
+            try:
+                config = self.persistence.get_config()
+                state = self.persistence.get_state()
+                tz = config.timezone or "UTC"
+                try:
+                    now = datetime.now(ZoneInfo(tz))
+                except Exception:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)